### ISSUE TITLE
Tutorial 0: button is huge and out of window

### ISSUE THREAD

**Issue by titimoby** [beeware/toga#336]
## Expected Behavior
As per the screenshot, I was waiting for a button in the window, with a proper size

## Current Behavior
The button is very huge and is out of the window

## Steps to reproduce
run tutorial 0 as explained in https://toga.readthedocs.io/en/latest/tutorial/tutorial-0.html

## Your Environment

* Python Version (list the specific version number)
Python 3.6.3

* Operating System and Version (select from the following and list the specific version number; if your OS is not listed, list that as well)

Ubuntu 17.10

* Toga Target (the type of app you are trying to generate)
 
Ubuntu desktop app

![tuto0](https://user-images.githubusercontent.com/348156/35342445-5f3f2b14-0128-11e8-8b41-281bb2a4c1e3.png)

**Comment by freakboy3742** [comment]
Thanks for the report - can you clarify the version of Toga that you're using (and how you installed it)? This might be a problem that has been resolved recently...

**Comment by titimoby** [comment]
How can I identify the version of Toga ?
I used pip install toga on my Ubuntu 17.10

**Comment by danyeaw** [comment]
I can't reproduce this issue in Ubuntu 17.10. To check the version of Toga run:

    pip show toga

The latest version is 0.3.0.dev4. You can install it with:

    pip install --pre toga

or from source with:

    pip install -e ./ src/core src/gtk src/dummy

**Comment by titimoby** [comment]
Sorry for the delay, unfortunately python apps are not my full time job ;)

```
(toga) thierry-laptop% pip show toga
Name: toga
Version: 0.2.15
Summary: A Python native, OS native GUI toolkit.
Home-page: http://pybee.org/toga
Author: Russell Keith-Magee
Author-email: russell@keith-magee.com
License: New BSD
Location: /home/thierry/projects/toga/lib/python3.6/site-packages
Requires: toga-gtk

```

**Comment by titimoby** [comment]
I uninstalled then installed 0.3.0.dev4
Result is better but still not what I though
![toga](https://user-images.githubusercontent.com/348156/35976631-02492a8a-0ce1-11e8-9601-0a55f1e7a0d1.png)

**Comment by freakboy3742** [comment]
Ok - while it may not look like it, this is better, and gets into territory that is slightly more explainable.

We've had other reports where the in-window menu isn't been considered as part of window geometry, but we haven't been able to narrow down the cause.

So - what other options do you have installed in Ubuntu? The default desktop for Ubuntu places menus on the top of the desktop, not the window.

**Comment by titimoby** [comment]
I'm using Ubuntu 17.10 Gnome
No specific options added.
I want to add that I appreciate the improvement with v0.3.0 I just fill comment here to bring help as much as I can. Thanks for all the good work.

**Comment by freakboy3742** [comment]
Taking another look at this - it looks like the problem might be that the vertical space that *should* be applied to the top of the main container is being applied *horizontally*. 

Visually, it looks like the button is too short on the right hand side by about the same amount that it needs to be pushed down vertically. I'm wondering if there's a simple x/y or width/height offset being mixed up somewhere in the GTK layout algorithm.

This would also explain why the geometry works fine when the menubar isn't contained in the window - the spacer is 0, and therefore has no effect.

**Comment by freakboy3742** [comment]
I think this is a duplicate of #139.
